comms/src/connection_manager/establisher.rs

```diff
@@ -332,6 +374,10 @@ where F: Fn(InprocAddress, usize) -> Result<(EstablishedConnection, NetAddress)>
                     _ => {},
                 }
             }
+            num_polls += 1;
+            if num_polls * 100 >= self.attempt_timeout.as_millis() {
```

Comment:
A variable/const can be used for the 100 here and the `monitor.read(100)` on ln 360 to ensure they get modified at the same time.